src/lib.rs

```diff
@@ -1110,6 +1134,49 @@ impl<'a, T> Iterator for Drain<'a, T> {
     }
 }
 
+/// This `struct` is created by [`HashBag::subtract`].
+/// See its documentation for more.
+pub struct Subtract<'a, T, S = RandomState> {
+    /// An iterator over `self` items
+    items: IterInner<'a, T>,
+
+    /// The bag with entries we DO NOT want to return
+    other: &'a HashBag<T, S>,
+}
+
+impl<'a, T: fmt::Debug> fmt::Debug for Subtract<'a, T> {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt.debug_struct("Subtract")
+            .field("items", &self.items)
+            .field("other", &self.other)
+            .finish()
+    }
+}
+
+impl<'a, T, S> Iterator for Subtract<'a, T, S>
+where
+    T: Eq + Hash,
+    S: BuildHasher,
+{
+    type Item = (&'a T, usize);
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        loop {
+            let (t, n) = self.items.next()?;
+            let other_n = self.other.contains(t);
+            if other_n < *n {
+                return Some((t, *n - other_n));
+            }
+        }
+    }
+
+    #[inline]
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.items.size_hint()
```

Comment:
We can only use the upper bound from the inner iterator, the lower bound should be 0: it is always possible every item in the iterator is also present in the other hash bag, and there will be zero items returned, even if there are guaranteed to be several items in the iterator left.

Reply:
I feel quite silly now. Good catch. It has been fixed now. Thank you.